src/resdk/tables/variant.py

```diff
@@ -79,90 +80,187 @@ class VariantTables(BaseTables):
     def __init__(
         self,
         collection: Collection,
-        discard_fakes: Optional[bool] = True,
-        mutations: Optional[List[str]] = None,
+        geneset: Optional[List[str]] = None,
+        filtering: bool = True,
         cache_dir: Optional[str] = None,
         progress_callable: Optional[Callable] = None,
     ):
         """Initialize class.
 
-        :param collection: collection to use
-        :param mutations: only consider these mutations
-        :param discard_fakes: discard fake variants
-        :param cache_dir: cache directory location, if not specified system specific
-                          cache directory is used
-        :param progress_callable: custom callable that can be used to report
-                                  progress. By default, progress is written to
-                                  stderr with tqdm
+        :param collection: Collection to use.
+        :param geneset: Only consider mutations from this gene-set.
+            Can be a list of gene symbols or a valid geneset Data
+            object id / slug.
+        :param filtering: Only show variants that pass QC filters.
+        :param cache_dir: Cache directory location, if not specified
+            system specific cache directory is used.
+        :param progress_callable: Custom callable that can be used to
+            report progress. By default, progress is written to stderr
+            with tqdm.
         """
         super().__init__(collection, cache_dir, progress_callable)
+        self.filtering = filtering
 
-        self.discard_fakes = discard_fakes
+        self._geneset = None
+        if geneset is None:
+            self._check_heterogeneous_mutations()
+            # Assign geneset from the Genialis Platform
+            self.geneset = self._get_obj_geneset(self._data[0])
+        else:
+            self.geneset = geneset
 
-        self.mutations = mutations
-        if mutations is not None:
-            if not isinstance(mutations, list):
-                raise ValueError("Mutations must be a list of strings")
+    @property
+    def geneset(self):
+        """Get geneset."""
+        return self._geneset
+
+    @geneset.setter
+    def geneset(self, value):
+        """Set geneset."""
```

Comment:
Add a type hint for value (int, str or Geneset if I am not mistaken).
Add some comment about the value attribute: it can either be id, slug or a Geneset instance.

Reply:
Also add comment that the geneset can be set only once and when ValueError is raised.

Reply:
Done.